src/stores/room-list/RoomListStore.ts

```diff
@@ -398,6 +420,21 @@ export class RoomListStoreClass extends AsyncStoreWithClient<IState> {
     }
 
     private async handleRoomUpdate(room: Room, cause: RoomUpdateCause): Promise<any> {
+        if (cause === RoomUpdateCause.NewRoom) {
+            if (this.holdNewRooms) {
+                console.log("Room updates are held: putting room " + room.roomId + " into the holding pen");
+                this.roomHoldingPen.push(room);
+                return;
+            } else {
+                // we call straight out to VoipUserMapper here which is a bit of a hack: probably this
+                // should be calling the visibility provider which in turn farms out to various visibility
+                // providers? Anyway, the point of this is that we delay doing anything about this room
+                // until the VoipUserMapper had had a chance to do the things it needs to do to decide
+                // if we should show this room or not.
+                await VoipUserMapper.sharedInstance().onNewInvitedRoom(room);
```

Comment:
Hmm, yeah, this is somewhat oddly placed... Listening to an event for this or placing in a visibility provider would make a bit more sense to me.

Reply:
Have gone via the visibility provider - hopefully this is a bit more palatable